middle_end/flambda2/types/env/typing_env.ml

```diff
@@ -1226,38 +1226,39 @@ end = struct
       let module VA = Value_approximation in
       match ty with
       | Value descr -> (
+        let value_unknown = VA.Unknown K.value in
         match Type_descr.descr descr with
-        | Unknown | Bottom -> Value_unknown
+        | Unknown | Bottom -> value_unknown
         | Ok (Equals simple) ->
           Simple.pattern_match' simple
             ~const:(fun const -> VA.Value_const const)
-            ~var:(fun _ ~coercion:_ -> VA.Value_unknown)
+            ~var:(fun var ~coercion:_ -> VA.Unknown (Variable.kind var))
```

Comment:
If I'm not mistaken, this should always be `value_unknown`. Should we put a check here, or is it not necessary?